src/NBitcoin/Network.cs

```diff
@@ -239,16 +239,10 @@ public byte[] MagicBytes
         /// </code>
         /// </example>
         /// <returns>A genesis block.</returns>
-        public static Block MineGenesisBlock(ConsensusFactory consensusFactory, string coinbaseText, Target target, Money genesisReward, int version = 1)
+        public static Block MineGenesisBlock(Network network, string coinbaseText, Target target, Money genesisReward, int version = 1)
         {
-            if (target == null)
-                throw new ArgumentException($"Parameter '{nameof(target)}' cannot be null. Example use: new Target(new uint256(\"0000ffff00000000000000000000000000000000000000000000000000000000\"))");
-
-            if (consensusFactory == null)
-            {
-                throw new ArgumentException($"Parameter '{nameof(consensusFactory)}' cannot be null. Use 'new ConsensusFactory()' for Bitcoin-like proof-of-work blockchains" +
-                                            "and 'new PosConsensusFactory()' for Stratis-like proof-of-stake blockchains.");
-            }
+            if (network == null)
```

Comment:
In fact this methos needs to be changed back because when it is used, we don't have a Network yet.
All the user can pass is `new PosConsensusFactory()` or similar.

Reply:
This has been reverted 👍